firestore/src/common/settings.cc

```diff
@@ -44,18 +47,66 @@ std::string ToStr(int64_t v) {
 }  // namespace
 
 #if !defined(__APPLE__)
-Settings::Settings() : host_(kDefaultHost) {}
+Settings::Settings()
+    : host_(kDefaultHost),
+      local_cache_settings_(std::make_shared<PersistentCacheSettings>(
+          PersistentCacheSettings::Create())) {}
 #endif
 
 void Settings::set_host(std::string host) { host_ = std::move(host); }
 
 void Settings::set_ssl_enabled(bool enabled) { ssl_enabled_ = enabled; }
 
+const LocalCacheSettings& Settings::local_cache_settings() {
+  if (cache_settings_source_ == CacheSettingsSource::kOld) {
+    if (is_persistence_enabled()) {
+      local_cache_settings_ = std::make_shared<PersistentCacheSettings>(
```

Comment:
It seems pretty weird to mutate a member variable (i.e. `local_cache_settings_`) in this "getter" member function. Could you find a way to instead make this member function `const` and initialize `local_cache_settings_` when it is mutated?